src/libraries/Microsoft.PowerFx.Connectors/ConnectorFunction.cs

```diff
@@ -359,19 +327,37 @@ internal ServiceFunction GetServiceFunction(string ns = null, HttpMessageInvoker
         internal async Task<FormulaValue> InvokeAync(FormattingInfo context, HttpClient httpClient, FormulaValue[] values, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
-            return await GetServiceFunction(null, httpClient).InvokeAsync(context, values, cancellationToken).ConfigureAwait(false);
+            return await GetServiceFunction(httpClient, ConnectorSettings).InvokeAsync(context, values, cancellationToken).ConfigureAwait(false);
+        }    
+
+        public Task<FormulaValue> InvokeAync(HttpClient httpClient, FormulaValue[] values, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            return InvokeAync(FormattingInfoHelper.CreateFormattingInfo(), httpClient, values, cancellationToken);
         }
 
-        public Task<FormulaValue> InvokeAync(IRuntimeConfig config, HttpClient httpClient, FormulaValue[] values, CancellationToken cancellationToken)
+        // For future implementation, we want to support the capability to run intellisense on optional parameters and need to use the parameter name to get the suggestions.
+        // An example is Office 365 Outlook connector, with GetEmails function, where folderPath is optional and defines x-ms-dynamic-values
+        // Also on SharePoint action SearchForUser (Resolve User)
+        //public async Task<ConnectorSuggestions> GetConnectorSuggestionsAsync(HttpClient httpClient, NamedFormulaType[] inputParams, string suggestedParamName, CancellationToken cancellationToken)
+
+        // This API only works on required paramaters and assumes we interrogate param number N+1 if N parameters are provided.
+        public async Task<ConnectorSuggestions> GetConnectorSuggestionsAsync(HttpClient httpClient, FormulaValue[] knownParameters, IServiceProvider services, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
-            return InvokeAync(config.ServiceProvider.GetFormattingInfo(), httpClient, values, cancellationToken);
+            ConnectorSettings connectorSettings = ConnectorSettings.Clone(throwOnError: false);            
+
+            ServiceFunction serviceFunction = GetServiceFunction(httpClient, connectorSettings);
+            return await serviceFunction.GetConnectorSuggestionsAsync(knownParameters, knownParameters.Length, services, cancellationToken).ConfigureAwait(false);
         }
 
-        public Task<FormulaValue> InvokeAync(HttpClient httpClient, FormulaValue[] values, CancellationToken cancellationToken)
+        public async Task<FormulaType> GetConnectorReturnSchemaAsync(HttpClient httpClient, FormulaValue[] knownParameters, IServiceProvider services, CancellationToken cancellationToken)
```

Comment:
FormulaType is not sufficient, as we also want to retrieve the visibility of the fields. We should have ConnectorParameterType or some other maybe new contract

Reply:
The FormulaType is coming from the interpretation of the JSON response and there is no 'visibility' in it.
Let's sync offline on this